src/AbstractApplication.php

```diff
@@ -62,6 +67,29 @@ public function __construct(Input $input = null, Registry $config = null)
 		$this->input = $input instanceof Input ? $input : new Input;
 		$this->config = $config instanceof Registry ? $config : new Registry;
 
+		// Create the container
```

Comment:
NO!  The application creating its own dependencies and filling the container in the way you have proposed here goes against EVERY practice regarding dependency injection and separation of concern.  Having the application be aware of the container in itself is not a major issue.  What is a major issue is the application class being hardcoded to instantiate its own container, fill it with its own keys and values, and being structured to hide all of the class' dependencies in the container, all in a way that cannot be overridden without essentially writing your own application classes.